tracing-error/src/lib.rs

```diff
@@ -74,8 +71,59 @@
 //!     }
 //! }
 //! ```
-//! In the future, this crate may also provide its own `Error` types as well,
-//! for users who do not wish to use other error-handling libraries.
+//!
+//! This crate also provides the [`InstrumentResult`]/[`InstrumentError`] traits which can be used
+//! to wrap errors with a [`TracedError`] which bundles the inner error with a [`SpanTrace`].
+//!
+//! ```rust
+//! # use std::error::Error;
+//! use tracing_error::InstrumentResult;
+//!
+//! # fn fake_main() -> Result<(), Box<dyn Error>> {
+//! std::fs::read_to_string("myfile.txt").in_current_span()?;
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! Once an error has been wrapped with with a [`TracedError`] the [`SpanTrace`] can be extracted
+//! one of 3 ways, either via [`TracedError`]'s `Display`/`Debug` implementations, or via the
+//! [`ExtractSpanTrace`] trait.
+//!
+//! ```rust
+//! use std::error::Error;
+//! use tracing_error::ExtractSpanTrace as _;
+//!
+//! fn print_extracted_spantraces(error: &(dyn Error + 'static)) {
+//!     let mut error = Some(error);
+//!     let mut ind = 0;
+//!
+//!     eprintln!("Error:");
+//!
+//!     while let Some(err) = error {
+//!         if let Some(spantrace) = err.span_trace() {
+//!             eprintln!("found a spantrace!:\n{}", spantrace);
+//!         } else {
+//!             eprintln!("{:>4}: {}", ind, err);
+//!         }
+//!
+//!         error = err.source();
+//!         ind += 1;
+//!     }
+//! }
+//!
+//! fn print_naive_spantraces(error: &(dyn Error + 'static)) {
+//!     let mut error = Some(error);
+//!     let mut ind = 0;
+//!
+//!     eprintln!("Error:");
+//!
+//!     while let Some(err) = error {
+//!         eprintln!("{:>4}: {}", ind, err);
+//!         error = err.source();
+//!         ind += 1;
+//!     }
+//! }
+//! ```
```

Comment:
I think it might be good if this one long example was split into two examples, with text in between explaining the two different approaches?